bake/compose.go

```diff
@@ -149,7 +150,6 @@ func (t *Target) composeExtTarget(exts map[string]interface{}) error {
 					}
 				}
 			case "cache-from":
-				t.CacheFrom = []string{} // Needed to override the main field
```

Comment:
unrelated but what is this `val.([]interface{})` `res.(string)` construction. Looks like asking for panic.

Reply:
> Needed to override the main field

Can you explain why we are reverting this?

Reply:
I see where it was discussed to replace the main field if it was defined for `cache-from`: https://github.com/docker/buildx/pull/721#discussion_r687950699. I think it makes sense to merge both fields. Do you have some concerns about it (expect the breaking change)?

> unrelated but what is this `val.([]interface{})` `res.(string)` construction. Looks like asking for panic.

https://github.com/docker/buildx/pull/1167